### ISSUE TITLE
False positive in unused_closure_parameter

### ISSUE THREAD

**Issue by jpsim** [realm/SwiftLint#1979]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
$ swiftlint lint --path test.swift
Linting Swift files at path test.swift
Linting 'test.swift' (1/1)
/Users/jsimard/Downloads/test.swift:3:5: warning: Unused Closure Parameter Violation: Unused parameter "manager" in a closure should be replaced with _. (unused_closure_parameter)
Done linting! Found 1 violation, 0 serious in 1 file.
```

### Environment

* SwiftLint version (run `swiftlint version` to be sure)?

0.24.0

* Installation method used (Homebrew, CocoaPods, building from source, etc)?

Homebrew

* Paste your configuration file:

None.

* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)?
  If so, paste their relative paths and respective contents.

None.

* Which Xcode version are you using (check `xcode-select -p`)?

/Applications/Xcode-9.2.app/Contents/Developer

* Do you have a sample that shows the issue? Run `echo "[string here]" | swiftlint lint --no-cache --use-stdin --enable-all-rules`
  to quickly test if your example is really demonstrating the issue. If your example is more
  complex, you can use `swiftlint lint --path [file here] --no-cache --enable-all-rules`.

```swift
import Foundation

({ (manager: FileManager) in
  print(manager)
})(FileManager.default)
```

Also, as reported by @irace:

![image](https://user-images.githubusercontent.com/474794/34273368-067ed370-e662-11e7-8845-4546527a6260.png)

**Comment by irace** [comment]
What customer service – thanks JP!

**Comment by jpsim** [comment]
So SourceKit is returning some BS with closures of this format. Here's what it thinks the _name_ of the closure is: `({ (manager: FileManager) in\n  print(manager)\n})`, and the body is `(FileManager.default)`.

Looking into making a workaround.

**Comment by marcelofabri** [comment]
This would be way easier/more reliable if we SourceKit returned closures in the structure (https://bugs.swift.org/browse/SR-6116). I started implementing it a few months ago, but it looks like I lost my WIP branch ¯\_(ツ)_/¯
